### PR TITLE
Add homework 04 of Stelian Todorichkov 18

### PR DESCRIPTION
**Pull request by StelianRBG**

### CODE REVIEW

File: homework_04/18_Stelian_Todorichkov/test_circle.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    #common ceter
+    circle1 = Circle(Point(0.0, 0.0), 3.0)
+    circle2 = Circle(Point(0.0, 0.0), 9.0)
+    assert circle1.find_relative_position(circle2) == RealativePosition.NO_COMMON_POINTS
+
+    #completly inside
+    circle1 = Circle(Point(0.0, 0.0), 9.0)
+    circle2 = Circle(Point(1.0, 0.0), 3.0)
+    assert circle1.find_relative_position(circle2) == RealativePosition.NO_COMMON_POINTS
```
Comment (Vic-Dim) [inline homework_04/18_Stelian_Todorichkov/test_circle.py:53]: It's nice that you've added several comments that specify the test cases

**Review by Vic-Dim — APPROVED** [review]

File: .github/workflows/homework_04_18_stelian_todorichkov.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    branches: [ main ]
+    paths: 
+      - homework_04/18_stelian_todorichkov/**
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8", "3.9"]
```
Comment (Lilly7777) [inline .github/workflows/homework_04_18_stelian_todorichkov.yaml:19]: You can also add the latest version of Python - 3.10

File: homework_04/18_stelian_todorichkov/circle.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        assert isinstance(other, Circle)
+        distance = self.center.distane_between_points(other.center)
+        if self == other:
+            return RealativePosition.SAME
+        if abs(self.radius - other.radius) < distance < (self.radius + other.radius):
+            return RealativePosition.INTERSECTING
+        if distance == (self.radius + other.radius) \
+                or distance == abs(self.radius - other.radius):
+            return RealativePosition.TOUCHING
+        return RealativePosition.NO_COMMON_POINTS
```
Comment (Lilly7777) [inline homework_04/18_stelian_todorichkov/circle.py:69]: The code is well structured, readable, with the necessary comments and cover pylint requirements.

File: homework_04/18_stelian_todorichkov/test_circle.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    circle1 = Circle(Point(-1.0, 1.0), 3.0)
+    circle2 = Circle(Point(3.0, -2.0), 2.0)
+    assert circle1.find_relative_position(circle2) == RealativePosition.TOUCHING
+
+    #inside
+    circle1 = Circle(Point(-1.0, 0.0), 3.0)
+    circle2 = Circle(Point(3.0, -3.0), 8.0)
+    assert circle1.find_relative_position(circle2) == RealativePosition.TOUCHING
+
+def test_no_common_points():
```
Comment (Lilly7777) [inline homework_04/18_stelian_todorichkov/test_circle.py:36]: I really like the comments on the fourth test - they help me understand your logic.

**Pull request merged**